test/e2e/test_job_submissions.py

```diff
@@ -842,6 +850,221 @@ def sync_input_actions_are_canceled(sessions: List[Dict[str, Any]]) -> bool:
+                                }
+                            ]
+                        },
+                        "dependencies": [{"dependsOn": step_one_name}],
+                        "script": {
+                            "actions": {
+                                "onRun": {
+                                    "command": (
+                                        "/bin/sleep"
+                                        if os.environ["OPERATING_SYSTEM"] == "linux"
+                                        else "powershell"
+                                    ),
+                                    "args": (
+                                        ["120"]
+                                        if os.environ["OPERATING_SYSTEM"] == "linux"
+                                        else ["ping", "localhost", "-n", "120"]
+                                    ),
+                                    "cancelation": {
+                                        "mode": "NOTIFY_THEN_TERMINATE",
+                                        "notifyPeriodInSeconds": 1,
+                                    },
+                                },
+                            }
+                        },
+                    },
+                    {
+                        "name": step_three_name,
+                        "hostRequirements": {
+                            "attributes": [
+                                {
+                                    "name": "attr.worker.os.family",
+                                    "allOf": [os.environ["OPERATING_SYSTEM"]],
+                                }
+                            ]
+                        },
+                        "dependencies": [{"dependsOn": step_two_name}],
+                        "script": {
+                            "actions": {
+                                "onRun": {"command": "whoami"},
+                            }
+                        },
+                    },
+                ],
+            },
+        )
+
+        # Wait for the job to start
+
+        @backoff.on_predicate(
+            wait_gen=backoff.constant,
+            max_time=120,
+            interval=10,
+        )
+        def is_job_started_with_sessions(current_job: Job) -> bool:
+            current_job.refresh_job_info(client=deadline_client)
+            LOG.info(f"Waiting for job {current_job.id} to be created")
+            if current_job.lifecycle_status == "CREATE_IN_PROGRESS":
+                return False
+            sessions: list[dict[str, Any]] = deadline_client.list_sessions(
+                farmId=job.farm.id, queueId=job.queue.id, jobId=job.id
+            ).get("sessions")
+            if sessions and len(sessions) > 0:
+                return True
+            return False
+
+        assert is_job_started_with_sessions(job)
+
+        # Wait some time for the second step (which sleeps for 2 minutes) to start
+        time.sleep(20)
```

Comment:
Instead of sleeping here, it may be worth just waiting on the second step to switch to `RUNNING`? That way if it starts sooner, the test can complete faster.

Reply:
That's true, but that's also more API calls. There's a tradeoff, and I believe that 20 seconds is not a big time wait to justify the extraneous API calls.

Reply:
I see the `sleep` command more as a last resort. It guarantees to slow down all testing, especially when testing new changes before PR. Between these two sleeps, you're adding thirty seconds to every full test attempt, which already takes an extremely long time to complete. If I'm not mistaken, most people make sure their changes work with the full suite more than once before creating a PR, that's at least a minute to every future change people try to merge.